db/dbtypes/types.go

```diff
@@ -1831,7 +1831,8 @@ func ReduceAddressHistory(addrHist []*AddressRow) (*AddressInfo, float64, float6
 	}
 
 	var received, sent, fromStake, toStake int64
-	var transactions, creditTxns, debitTxns []*AddressTx
+	var creditTxns, debitTxns []*AddressTx
+	transactions := make([]*AddressTx, 0, len(addrHist))
```

Comment:
There's the possibility this will over-allocate because of `!addrOut.ValidMainChain`, but the vast majority of most transactions are on stakeholder-approved main chain blocks, so this is still a good optimization.